superset/assets/src/SqlLab/components/ResultSet.jsx

```diff
@@ -64,6 +64,14 @@ export default class ResultSet extends React.PureComponent {
       this.fetchResults(nextProps.query);
     }
   }
+  componentDidUpdate() {
```

Comment:
I don't think this is the right place to update query state. this will cause whole component rerender.

Reply:
I tried, `success` state is unnecessary for sqllab to show query results. However, if you navigate to `Query History`, the state will show `rendering` instead of success. That's why I add this `componentDidUpdate()`.